data/transactions/logic/eval.go

```diff
@@ -563,7 +563,10 @@ type frame struct {
 	returns int
 }
 
-type scratchSpace [256]stackValue
+// ScratchSpaceSize is the length of all scratch slot (scratch space), const 256
+const ScratchSpaceSize = 256
+
+type scratchSpace [ScratchSpaceSize]stackValue
```

Comment:
Total nit: you could just make `EvalContext.scratch` public instead of this change. We already have a public stack, so there is a precedent

Reply:
good point, it turned out that `scratch` is not as widely used as `stack`, so we can easily rename it, and the query method can be removed too.